lambda/listRecordsForInterval.ts

```diff
@@ -0,0 +1,50 @@
+import {
+	QueryCommand,
+	ValidationException,
+	type QueryCommandOutput,
+	type TimestreamQueryClient,
+} from '@aws-sdk/client-timestream-query'
+import { parseResult } from '@bifravst/timestream-helpers'
+import { getQueryString } from './getQueryString.js'
+import type { HistoricalDataTimeSpan } from './historicalDataTimeSpans.js'
+
+type historyRecordReturnType = {
+	ICCID: string
+	ID: string
+	measure_name: string
+	time: Date
+	'measure_value::double': number
+}
+
+export const listRecordsForInterval =
+	(ts: TimestreamQueryClient, dbName: string, tableName: string) =>
+	async ({
+		timespan: { binIntervalMinutes, durationHours },
+		iccid,
+	}: {
+		timespan: Pick<
+			HistoricalDataTimeSpan,
+			'binIntervalMinutes' | 'durationHours'
+		>
+		iccid: string
+	}): Promise<historyRecordReturnType[] | null> => {
+		const QueryString = getQueryString({
+			timespan: { binIntervalMinutes, durationHours },
+			iccid,
+			dbName,
+			tableName,
+		})
+		let result
+		try {
+			result = await ts.send(
+				new QueryCommand({
+					QueryString,
+				}),
+			)
+		} catch (err) {
+			if (err instanceof ValidationException) {
```

Comment:
This would shadow an actual error in your SQL, so instead of catching this, make sure to only query the available columns.

You can get the available columns by doing a `SELECT * FROM table LIMIT 1`, and through that determine which columns are present.

See https://github.com/hello-nrfcloud/backend/blob/19a9e47d54cbc7404017e1bd335570cad8c25f66/historicalData/getAvailableColumns.ts#L6
and https://github.com/hello-nrfcloud/backend/blob/19a9e47d54cbc7404017e1bd335570cad8c25f66/lambda/queryLwM2MHistory.ts#L122-L133